packages/http/src/responses.ts

```diff
@@ -140,7 +140,8 @@ function getResponseStatusCodes(
     }
   }
 
-  return codes;
+  // eliminate duplicate status codes
+  return [...new Set(codes)];
```

Comment:
Wondering if we shouldn't instead be giving an error(or warning) that you have duplicate status code. Any maybe also if you have 2 properties with `@statusCode` which was the original cause of this issue.

Reply:
Like https://cadlplayground.z22.web.core.windows.net/?c=aW1wb3J0ICJAdHlwZXNwZWMvaHR0cCI7Cgp1c2luZyBUeXBlU3BlYy5IdHRwOwoKb3AgcmVhZCgpOiB7CiAgQHN0YXR1c0NvZGUgXzogMjAwO8QWYm9keSDEBcYmzihzxCgxxCggIG5hbWU6IHN0cmluZ8QSfTsKfTsK&e=@typespec/openapi3&options=%7B%22linterRuleSet%22:%7B%22extends%22:%5B%22@typespec/http/all%22%5D%7D%7D

Also if you name both properties the same it somehow doesn't have the duplication

Reply:
Based on your original repro:
```
model CustomUnauthorizedResponse {
  @statusCode _: 401;
  @body body: UnauthorizedResponse;
}

op list(): CustomUnauthorizedResponse;
```

I don't see how we would want to emit an error or warning here unless we are saying that there shouldn't be a `@statusCode` annotation on `body` because it is on `UnauthorizedResponse`.

Reply:
I mean yeah I feel like that shouldn't be allowed really, the exampled I shared above is the same but here there is named models. UnauthorizedResponse here is not meant to be used as a body it only include a statusCode so it feels like the wrong use of that model anyway

that model is basically just adding a statusCode

```
model UnauthorizedResponse is Response<401>;

model Response<Status> {
  @doc("The status code.")
  @statusCode
  statusCode: Status;
}
```


A more simple example that makes even less sense:
```
op read(): {
  @statusCode _: 200;
  @statusCode s: 201;
};
```
here we are saying that response has 2 status code at once. It shouldn't meant that its either 200 or 201
https://cadlplayground.z22.web.core.windows.net/?c=aW1wb3J0ICJAdHlwZXNwZWMvaHR0cCI7Cgp1c2luZyBUeXBlU3BlYy5IdHRwOwoKb3AgcmVhZCgpOiB7CiAgQHN0YXR1c0NvZGUgXzogMjAwO88Wc8QWMTsKfTsK&e=@typespec/openapi3&options=%7B%22linterRuleSet%22:%7B%22extends%22:%5B%22@typespec/http/all%22%5D%7D%7D

Reply:
So, there are unit tests in place actual codifying this as by-design:

```
 model CreatedOrUpdatedResponse {
        @statusCode ok: "200";
        @statusCode created: "201";
      }
```

The commit that added this: https://github.com/microsoft/typespec/commit/ffa4c71e3ec77d6832fac505e74666e6b7de63a7

Reply:
That seems weird if I put my code generator hat on. What code would be generated and how would I access the status code from such a model?

Reply:
Per the design discussion, the plan is to make having duplicate status codes an error in the HTTP library.